pkg/exchange/max/exchange.go

```diff
@@ -210,16 +210,30 @@ func (e *Exchange) QueryOrderTrades(ctx context.Context, q types.OrderQuery) ([]
 }
 
 func (e *Exchange) QueryOrder(ctx context.Context, q types.OrderQuery) (*types.Order, error) {
-	if q.OrderID == "" {
-		return nil, errors.New("max.QueryOrder: OrderID is required parameter")
+	if len(q.OrderID) == 0 && len(q.ClientOrderID) == 0 {
+		return nil, errors.New("max.QueryOrder: one of OrderID/ClientOrderID is required parameter")
 	}
 
-	orderID, err := strconv.ParseInt(q.OrderID, 10, 64)
-	if err != nil {
-		return nil, err
+	if len(q.OrderID) != 0 && len(q.ClientOrderID) != 0 {
+		return nil, errors.New("max.QueryOrder: only accept one parameter of OrderID/ClientOrderID")
+	}
+
+	request := e.v3order.NewGetOrderRequest()
+
+	if len(q.OrderID) != 0 {
+		orderID, err := strconv.ParseInt(q.OrderID, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+
+		request.Id(uint64(orderID))
+	}
+
+	if len(q.ClientOrderID) != 0 {
```

Comment:
you need `else if` here?

Reply:
No, because we already exclude the situation 'q.OrderID == 0 && q.ClientOrderID == 0'.